aea/cli/utils/decorators.py

```diff
@@ -100,25 +106,27 @@ def _validate_config_consistency(ctx: Context):
                 )
             )
 
-        _check_aea_version(package_configuration)
+        if check_aea_version:
+            _check_aea_version(package_configuration)
         _compare_fingerprints(
             package_configuration, package_directory, is_vendor, item_type
         )
 
 
-def _check_aea_project(args):
+def _check_aea_project(args, check_aea_version: bool = True):
     try:
         click_context = args[0]
         ctx = cast(Context, click_context.obj)
         try_to_load_agent_config(ctx)
         skip_consistency_check = ctx.config["skip_consistency_check"]
         if not skip_consistency_check:
-            _validate_config_consistency(ctx)
+            _validate_config_consistency(ctx, check_aea_version=check_aea_version)
     except Exception as e:  # pylint: disable=broad-except
         raise click.ClickException(str(e))
 
 
-def check_aea_project(f):
+@decorator_with_optional_params
+def check_aea_project(f, check_aea_version: bool = True):
```

Comment:
now we can use either

```python
@check_aea_project
def cli_function
```

or


```python
@check_aea_project(*args, **kwargs)
def cli_function
```

Reply:
If the wrapper has some positional arguments other than the function being wrapped, we can only use the latter version.

Reply:
Basically, all of this was to avoid to update `@check_aea_project` with `@check_aea_project()` everywhere, just because of the added parameter `check_aea_version`.

To avoid the updates, it could have been done in another way, by using `partial` on `check_aea_version`, instead of this machinery. But this is a more general solution that we can reuse.